Code-Sleep-Python/8-Puzzle game/puzzle8.py

```diff
@@ -0,0 +1,331 @@
+		                |                       |
+		                |                       |
+		                |                       |
+		                |=======================|
+
+
+
+
+
+
+		        	Press \u21b3 (ENTER Key) to start new Puzzle
+
+
+
+
+
+
+
+		"""
+
+	def printControls(self):
+		return """
+		            \u2191  -> Move Up
+		            \u2193  -> Move Down
+		            \u2190  -> Move Left
+		            \u2192  -> Move Right
+		             x  -> Declare Non Solvable
+		             n  -> New Puzzle
+		             a  -> Automate
+		"""
+
+	def totalInversions(self):
+		"""
+		odd no of inversions means problem is not solvable
+		"""
+		inversions = 0
+		nboard = self.board.copy()
+		nboard.remove(' ')
+		for i in range(0,8):
+			for j in range(i+1,8):
+				if nboard[i]>nboard[j]:
+					inversions+=1
+		return inversions
+
+
+	def in_priority_queue(self,count):
+		"""
+		count is used in case 1st entry of tuple is equal
+		priorityqueue uses 2nd entry of tuple to sort, in case 1st entry is equal
+		third entry in tuple can't be compared to sort entries thats why we are using count
+		"""
+		return (self.manhattendistance()+self.moves,count,self)
+
+
+def automate(board,window):
+	queue = PriorityQueue()
+	
+	#odd inversion is not solvable
+	if board.totalInversions()%2==1:
+		window.clear()
+		window.insstr(0,0,
+			"""
+			____________________________
+			|  Problem is not Solvable |
+			|     (Odd Inversions)     |	
+			----------------------------
+			""")
+		window.refresh()
+		time.sleep(1)
+		window.getch()
+		return 
+	queue.put(board.in_priority_queue(0))
+	path = []
+	i=1
+	while not queue.empty():
+		bestboard = queue.get()[2]
+		if not bestboard.isSolution():
+			for neighbor in bestboard.getNeighbors():
+				if neighbor != bestboard.previous:
+					queue.put(neighbor.in_priority_queue(i))
+					i+=1
+		else:
+			path.append(bestboard)
+			prev = bestboard.previous
+			while prev is not None:
+				path.append(prev)
+				prev = prev.previous
+			break
+
+	path.reverse()
+	for board in path:
+		window.clear()
+		window.insstr(0,0,str(board))
+		window.refresh()
+		time.sleep(1.5)
+
+	window.insstr(18, 0, "\t\tPress \u21b3 (ENTER Key) to Continue" )
+	window.refresh()
+	window.getch()
+
+def initialize(window):
+	board = Puzzle()
+	window.insstr(0, 0, str(board))
+	window.insstr(18,0,board.printControls())
+	ch = window.getch()
+	while str(ch)!='10':
+		if ch==curses.KEY_UP:
+			board.move_tile("up");
+		elif ch==curses.KEY_DOWN:
+			board.move_tile("down")
+		elif ch==curses.KEY_LEFT:
+			board.move_tile("left")
+		elif ch==curses.KEY_RIGHT:
+			board.move_tile("right")
+		elif ch==ord("n"):
+			board =Puzzle(None)
+		elif ch == ord("x"):
+			window.clear()
+			if board.totalInversions()%2==1:
+				window.insstr(0,0,board.getvictorySign())
+				board = Puzzle()
+			else:
+				window.insstr(0,0,
+				"""
+			____________________________
+			|   Problem is solvable!!  |
```

Comment:
keep this in a separate parameter and pass through the function.
Same as above instances